### DISCUSSION TITLE
Problem with treesitter handling of errors

### DISCUSSION THREAD

**Discussion by Lattay** [neovim/neovim#29585] [Q&A]
Hi,

I have been noticing several time that when treesitter encounter an error in highlight neovim becomes really painfull to interact with. Essentially it triggers something like a lua `error` call (triggers a press enter prompt) but on every keystroke!

The problem is that it is not particularly rare. It often happen when the runtime files have been updated but the TS parser has not for example.

Today I have something wrong with latex highlight queries and updating everything doesn't help. The only solution I have right now to get on with my work (because I don't want to debug neovim today) is to disable TS, which is fairly anoying when you have to presse enter between each key.

So the questions:
- Is there a way to make these errors less invasive?
- Should this behavior be considered a bug?

**Comment by clason** [comment]
* Not easily, since these are thrown in a special "fast" decoration provider loop. We already worked to make them less intrusive; that is the best we can do so far. See https://github.com/neovim/neovim/issues/27235 for the tracking issue.
* It's not a bug but simply how tree-sitter works. Until upstream adds versioning to parsers and queries (which they have so far refused to consider), you just have to be very careful when adding treesitter based plugins and config. (Neovim itself is stable with respect to the features that are enabled by default, but the whole ecosystem very much is not. Be _very_ conservative when installing and enabling extra features.)
